dmoj/cptbox/tracer.py

```diff
@@ -87,51 +97,34 @@ def readstr(self, address, max_size=4096):
         return utf8text(read)
 
 
-# SecurePopen is a subclass of a cython class, _cptbox.Process. Since it is exceedingly unwise
-# to do everything in cython, determining the debugger class is left to do here. However, since
-# the debugger is constructed in __cinit__, we have to pass the determined debugger class to
-# SecurePopen.__new__. While we can simply override __new__, many complication arises from having
-# different parameters to __new__ and __init__, the latter of which is given the *original* arguments
-# as passed to type.__call__. Hence, we use a metaclass to pass the extra debugger argument to both
-# __new__ and __init__.
-class TracedPopenMeta(type):
-    def __call__(self, argv, executable=None, *args, **kwargs):
-        executable = executable or find_exe_in_path(argv[0])
-        arch = file_arch(executable)
-        debugger = _arch_map.get((INTERPRETER_ARCH, arch))
-        if debugger is None:
-            raise RuntimeError('Executable type %s could not be debugged on Python type %s' % (arch, INTERPRETER_ARCH))
-        return super().__call__(debugger, self.debugger_type, argv, executable, *args, **kwargs)
-
-
-class TracedPopen(Process, metaclass=TracedPopenMeta):
-    debugger_type = AdvancedDebugger
+class TracedPopen(Process):
+    def create_debugger(self):
+        return AdvancedDebugger(self)
 
     def __init__(
-        self,
-        debugger,
-        _,
-        args,
-        executable=None,
-        security=None,
-        time=0,
-        memory=0,
-        stdin=PIPE,
-        stdout=PIPE,
-        stderr=None,
-        env=None,
-        nproc=0,
-        fsize=0,
-        address_grace=4096,
-        data_grace=0,
-        personality=0,
-        cwd='',
-        fds=None,
-        wall_time=None,
+            self,
+            args,
+            avoid_seccomp=False,
```

Comment:
```suggestion
            use_seccomp=True,
```

Reply:
Disagree, because use_seccomp=True can still not use seccomp on non-seccomp builds.